engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
```

Comment:
+1 for Pramod's and Chandni's point. If you need custom checkpointing you use a different storage agent just like people have done with in memory checkpointing. If it is not easy or fast to use a Storage agent to do this then we should think about how to improve the Storage API, not introduce a hacky way of checkpointing. If an operator is Stateless that means it has no state.